app/jobs/digital_object_import_processing_job.rb

```diff
@@ -44,13 +44,18 @@ def self.perform(digital_object_import_id)
   end
 
   def self.digital_object_for_digital_object_data(digital_object_data)
-    # If a uid is present in the digital_object_data, then this is an update operation
-    return DigitalObject.find_by_uid!(digital_object_data['uid']) if digital_object_data['uid'].present?
-
-    # No uid present, so we'll create a new object with type based on the digital_object_type value
-    return Hyacinth::Config.digital_object_types.key_to_class(digital_object_data['digital_object_type']).new if digital_object_data['digital_object_type'].present?
-
-    raise ArgumentError, 'Unable to find or create digital object because neither "uid" nor "digital_object_type" attributes were present in digital_object_data.'
+    if digital_object_data['assign_uid'].present?
+      raise ArgumentError, 'Either "uid" or "assign_uid" attribute may be present in digital_object_data, but not both.' if digital_object_data['uid'].present?
+      raise ArgumentError, 'Cannot create new digital object with "assign_uid"; object with same uid already exists' if DigitalObject.find_by_uid(digital_object_data['assign_uid'])
+      raise ArgumentError, 'Cannot create new digital object with "assign_uid"; digital object type is required' unless digital_object_data['digital_object_type'].present?
+      # assign_uid present, so we'll create a new object with type based on the digital_object_type value
+      Hyacinth::Config.digital_object_types.key_to_class(digital_object_data['digital_object_type']).new
```

Comment:
If this line is inside of the `if digital_object_data['assign_uid'].present?` block, does that prevent creation of a new digital object in the case when assign_uid isn't present?  (i.e. the usual case of creating a digital object and letting the system auto-assign a new uid)